dali/pipeline/data/view_test.cc

```diff
@@ -21,7 +21,7 @@ namespace dali {
 
 TEST(TensorList, View) {
   TensorList<CPUBackend> tl;
-  std::vector<Dims> shapes = { { 640, 480, 3}, { 320, 240, 1 } };
+  kernels::TensorListShape<> shapes = { { kernels::TensorShape<>{640, 480, 3}, {320, 240, 1} } };
```

Comment:
How about adding a constructor with `std::initializer_list<TensorShape<sample_ndim>>` to `TensorListShape`? It should take precedence when actually using braced initializer and resolve the ambiguity.

Reply:
At the same time it would cause `TensorListShape<> tls{1, 2}` to mean `TensorListShape<> tls(std::initializer_list(TensorShape<>{1}, TensorShape<>{2}))` instead of calling `TensorListShape<>::TensorListShape(int num_samples, int sample_dim)`

Reply:
Going through initializer_list<vector<...>>, which does not have non-explicit constructor for inty arguments solves both of the problems (and probably creates some new ones).